<a href="https://colab.research.google.com/github/guldenizcanatan/A-B-TEST/blob/main/a_b_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)

control = pd.read_excel('/content/ab_testing_data.xlsx', sheet_name='Control Group')
test = pd.read_excel('/content/ab_testing_data.xlsx', sheet_name='Test Group')


In [2]:
control.head()

Impression        Click    Purchase      Earning
0   82529.459271  6090.077317  665.211255  2311.277143
1   98050.451926  3382.861786  315.084895  1742.806855
2   82696.023549  4167.965750  458.083738  1797.827447
3  109914.400398  4910.882240  487.090773  1696.229178
4  108457.762630  5987.655811  441.034050  1543.720179

In [3]:
test.head()

Impression        Click    Purchase      Earning
0  120103.503796  3216.547958  702.160346  1939.611243
1  134775.943363  3635.082422  834.054286  2929.405820
2  107806.620788  3057.143560  422.934258  2526.244877
3  116445.275526  4650.473911  429.033535  2281.428574
4  145082.516838  5201.387724  749.860442  2781.697521

95% Confidence Interval For Purchases 

In [4]:
import statsmodels.stats.api as sms

sms.DescrStatsW(control["Purchase"]).tconfint_mean()


/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


(508.0041754264924, 593.7839421139709)

In [5]:
sms.DescrStatsW(test["Purchase"]).tconfint_mean()

(530.5670226990062, 633.6451705979289)

Purchases in the control group are between 508 and 593 with 95% confidence. Purchases in the test group are between 530 and 633 with 95% confidence

A/B Testing
first, are the data normally distributed or are the variances homogeneous?
then T test or nonparametric test


In [6]:
# 1.1 Normallity Assumption
############################

# H0: Purchases in Control follows normal distribution.
# H1: Purchases in Control Group do not follows normal distribution.
# H0: Purchases in Test follows normal distribution.
# H1: Purchases in Test Group do not follows normal distribution.

from scipy.stats import shapiro


test_statistic, pvalue = shapiro(control["Purchase"])
print('Test Statistic = %.4f, p-value = %.4f' % (test_statistic, pvalue))

#If p-value < 0.05  we reject null hypothesis
# p-value >  0.05 we can not reject null hypothesis
# assumptions are provided for the control group.

test_statistic, pvalue = shapiro(test["Purchase"])
print('Test Statistic = %.4f, p-value = %.4f' % (test_statistic, pvalue))

#If p-value < 0.05 we reject null hypothesis
# p-value > 0.05 we can not reject null hypothesis
# # assumptions are provided for the test group.


Test Statistic = 0.9773, p-value = 0.5891
Test Statistic = 0.9589, p-value = 0.1541


since p-value is greater than 0.05 for both groups
h0 cannot rejected, the assumption of normality is provided.

In [7]:
# 1.2 Homogeneity of Variances Assumption
############################

# H0: Variances are homogeneous
# H1: Variances are not homogeneous

from scipy import stats
stats.levene(control["Purchase"], test["Purchase"])


#If p-value <  0.05, H0 reject
# p-value >  0.05 H0 can not reject
# Variances are homogeneous.


LeveneResult(statistic=2.6392694728747363, pvalue=0.10828588271874791)

The variance is homogeneous because the p-value is greater than 0.05

*Since the assumption of normality and homogeneity is provided, we will perform an independent two-sample t-test.*

In [8]:
# 2. Proose a Hypothesis
############################


# H0: 𝜇1=𝜇2 (There is no statistically significant difference between the means of purchase in the control and test groups)
# H1: 𝜇1≠𝜇2 (There is a statistically significant difference between the means.)

test_statistic, pvalue = stats.ttest_ind(control["Purchase"], test["Purchase"], equal_var=True)

print('Test Statistic = %.4f, p-value = %.4f' % (test_statistic, pvalue))

# If p-value <  0.05 H0 reject
# If p-value >  0.05 H0 can not reject





Test Statistic = -0.9416, p-value = 0.3493


 p-value is greater than 0.05, there is no statistically significant difference between the two groups with 95% confidence.